### PR TITLE
Inherit directly from field & django 1.10 compatibility

### LINKED ISSUE #15: There is no way to loop over all superform bound fields
Composite fields not listed in the form.visible_fields and hidden_fields

Django suggest the form rendering customization by looping over `form.hidden_field` and `form.visible_fields` in the template - https://docs.djangoproject.com/ja/1.9/topics/forms/#looping-over-hidden-and-visible-fields. But the bound composite fields is not returned.

It seems it could be fixed but overriding `__iter__` method

```
    def __iter__(self):
        for name in self.fields:
            yield self[name]
       for name in self.composite_fields:
           yield self[name]
```

### PR DESCRIPTION
**Pull request by lucaswiman**
@gregmuellegger This PR significantly simplifies `django-superform` by having `FormSetField` and `FormField` inherit directly from `Field`, using the builtin [`Widget.value_from_datadict` API](https://docs.djangoproject.com/en/1.10/ref/forms/widgets/#django.forms.Widget.value_from_datadict). This has a number of advantages:
- It's much simpler and less code. This allows `SuperForm` to inherit directly from `Form` without need for a custom metaclass.
- It uses documented Django APIs, so should continue working more effectively on future versions of Django.
- Since composite fields are now just included in `form.fields`, this fixes #15 and removes the need for #16 (cc @kmmbvnr).

I also added support for Django 1.10 by updating the test environment and removing a few deprecated kwargs in template rendering. That broke compatibility with Django <1.8. Since all versions before 1.8 have hit end-of-life (and do not even receive security updates), I dropped support for them in the build matrix, along with Python 2.6.

Thanks again for writing this library! It's proven quite useful.

### CODE REVIEW

**Comment by gregmuellegger** [comment]
On the first glance, this is a super sweet change, removing the nastier parts of the implementation! Awesome :fireworks: 

I will try to dive deeper into it soonish. What might be worth adding before merging might be some kind of update-guide, or a description of the public-facing APIs that changed in the `CHANGES.rst` file.

File: django_superform/widgets.py [hunk 1]
```diff
@@ -47,17 +46,28 @@ def render(self, name, value, attrs=None, **kwargs):
...
+        return loader.render_to_string(template_name, context=context)
 
 
 class FormWidget(TemplateWidget):
     template_name = 'superform/formfield.html'
     value_context_name = 'form'
 
+    @property
+    def media(self):
+        return self.field.form_class().media
```
Comment (lucaswiman) [inline django_superform/widgets.py]: Instantiating the `form_class` / `formset_class` to get the media can cause some problems. I'll try to see if there's a way around it.

Comment (lucaswiman) [inline django_superform/widgets.py]: Addressed in 1beb80890178d4a0193a943f318290fa8f66e7bb.

**Comment by lucaswiman** [comment]
@gregmuellegger After attempting to integrate this into a real-world application, I think this needs some more work before it's ready for review. I'll comment when I'd like you to take another look. Apologies for the delay.

**Comment by gregmuellegger** [comment]
No hurries on that. The internals would be much nicer that way. Just ping me, I'll try to give it a look timely after notify me. Thanks for all the effort you put into this.

**Comment by lucaswiman** [comment]
I've updated the changelog, and I think this is ready for review. For the moment, this PR solves my immediate need of using superform with another forms library that has a conflicting custom metaclass, so it's good enough for now. 

To summarize the issues I was considering, using `value_from_datadict` allows the form/formset to show up in `cleaned_data`, but makes it hard to ensure that the values in `.forms` and `.formsets` are pointing to the _same_ instances. I think this branch is about as far as I can get on that front without removing the `forms`/`formsets` attributes, which would break backwards compatibility.

In my opinion, it is worth eventually breaking backwards compatibility (and possibly using other standard django mechanisms like `Widget.subwidgets()`) in the hopes of upstreaming this functionality to Django core. However, it might make sense to do that in a fork of this library, or as a prototype for inclusion inside Django.

**Comment by auvipy** [comment]
awesome changes. did you tried django 1.11b1 with this ?

**Comment by lucaswiman** [comment]
> awesome changes. did you tried django 1.11b1 with this ?

No. That's unrelated to this change and probably better addressed in a different PR.

**Comment by auvipy** [comment]
ok. but will be great to see this pr merged soon

**Comment by gregmuellegger** [comment]
Finally got around to review the PR thoroughly. I'm keen to know what you think on the topics brought up.

**Pull request closed without merging**